test/unit/disposable_common.dart

```diff
@@ -60,6 +60,29 @@ void testCommonDisposable(Func<StubDisposable> disposableFactory) {
       });
       disposable.dispose();
     });
+
+    test('should throw during disposal', () {
+      var completer = new Completer();
+      disposable.awaitBeforeDispose(completer.future);
+      disposable.dispose();
+      new Future(() {}).then((_) {
```

Comment:
That's what I thought, but I tested it and it seemed to work. Is it intermittent?